core/logging_test.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"testing"
+
+	"github.com/op/go-logging"
+	"github.com/spf13/viper"
+)
+
+func TestLoggingLevelDefault(t *testing.T) {
+	viper.Reset()
+
+	LoggingInit("")
+
+	assertDefaultLoggingLevel(t, DefaultLoggingLevel())
+}
+
+func TestLoggingLevelOtherThanDefault(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "warning")
+
+	LoggingInit("")
+
+	assertDefaultLoggingLevel(t, logging.WARNING)
+}
+
+func TestLoggingLevelForSpecificModule(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "core=info")
+
+	LoggingInit("")
+
+	assertModuleLoggingLevel(t, "core", logging.INFO)
+}
+
+func TestLoggingLeveltForMultipleModules(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "core=warning:test=debug")
+
+	LoggingInit("")
+
+	assertModuleLoggingLevel(t, "core", logging.WARNING)
+	assertModuleLoggingLevel(t, "test", logging.DEBUG)
+}
+
+func TestLoggingLevelForMultipleModulesAtSameLevel(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "core,test=warning")
+
+	LoggingInit("")
+
+	assertModuleLoggingLevel(t, "core", logging.WARNING)
+	assertModuleLoggingLevel(t, "test", logging.WARNING)
+}
+
+func TestLoggingLevelForModuleWithDefault(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "info:test=warning")
+
+	LoggingInit("")
+
+	assertDefaultLoggingLevel(t, logging.INFO)
+	assertModuleLoggingLevel(t, "test", logging.WARNING)
+}
+
+func TestLoggingLevelForModuleWithDefaultAtEnd(t *testing.T) {
+	viper.Reset()
+	viper.Set("logging_level", "test=warning:info")
+
+	LoggingInit("")
+
+	assertDefaultLoggingLevel(t, logging.INFO)
+	assertModuleLoggingLevel(t, "test", logging.WARNING)
+}
+
+func TestLoggingLevelForSpecificCommand(t *testing.T) {
```

Comment:
The current logging logic simply uses the command given to do a lookup in viper:
```Go
spec := viper.GetString("logging_level")
if spec == "" {
	spec = viper.GetString("logging." + command)
}
// Parse spec
```

Doing multiple different commands would just exercise the same logic but I can add some extra ones if you would like.

I think the logging level logic needs a look at which I was going to do for issue 802 but I thought it would be best to get some tests in place first.